app/src/common/shared/com/igalia/wolvic/ui/widgets/Windows.java

```diff
@@ -1033,6 +1041,27 @@ public void exitResizeMode() {
         }
     }
 
+    private final SharedPreferences.OnSharedPreferenceChangeListener mPreferencesListener = new SharedPreferences.OnSharedPreferenceChangeListener() {
+        @Override
+        public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+            if (!Objects.equals(key, mContext.getString(R.string.settings_key_window_distance)))
+                return;
+
+            WindowWidget frontWindow = getFrontWindow();
```

Comment:
Do we really need the fromWindow variable ?

Reply:
Actually yes, because `getFrontWindow()` can return null.

I have added the missing null-check.